hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/DispatchHandleContext.java

```diff
@@ -354,12 +353,19 @@ public Fees dispatchComputeFees(
                 authorizer,
                 storeFactory.asReadOnly(),
                 consensusNow,
-                transactionCategory == TransactionCategory.BATCH_INNER ? verifier : null,
-                transactionCategory == TransactionCategory.BATCH_INNER
+                shouldChargeForSigVerification(txBody) ? verifier : null,
+                shouldChargeForSigVerification(txBody)
                         ? txnInfo.signatureMap().sigPair().size()
                         : 0));
     }
 
+    private boolean shouldChargeForSigVerification(@NonNull final TransactionBody txBody) {
+        // The SystemContractGasCalculator's feeCalculator relies on the parent contract call context.
+        // The batchKey check is used to differentiate batch inner transactions from system contract calls
+        // dispatched by inner transactions.
+        return transactionCategory == TransactionCategory.BATCH_INNER && txBody.hasBatchKey();
+    }
+
```

Comment:
```suggestion
    private boolean isSubmittedInnerTransaction(@NonNull final TransactionBody txBody) {
        // Certain batch transactions can trigger child transactions inside the batch itself. Such child transactions 
        // must be verified with the parent transaction's context instead of the signatures on the child transaction.
        // We therefore need to differentiate contextual child transactions from the batch's submitted transaction
        // bodies themselves, which we'll do by checking the transaction body for an included batch key. 
        return transactionCategory == TransactionCategory.BATCH_INNER && txBody.hasBatchKey();
    }
```suggestion
    private boolean shouldChargeForSigVerification(@NonNull final TransactionBody txBody) {
        // The SystemContractGasCalculator's feeCalculator relies on the parent contract call context.
        // The batchKey check is used to differentiate batch inner transactions from system contract calls
        // dispatched by inner transactions.
        return transactionCategory == TransactionCategory.BATCH_INNER && txBody.hasBatchKey();
    }

```
```